### DISCUSSION TITLE
Missing Proposals / Specs

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/csharplang#4758] [Q&A] [answered]
Currently we have a lot of "placeholder proposals". Could we replace them with at least incomplete specification.
Looking at other sources like roslyn, docs.microsoft or drafts from e.g. Bill Wagner is really annoying.

Why are those propsoal approved without a spec - or where is the spec?

**Comment by 333fred** [comment, marked as answer]
The only things still missing are C# 7 features (and possibly one or two small C# 8 features), and specification work is ongoing for them on https://github.com/dotnet/csharpstandard/tree/draft-v7. For the ones that have nothing yet, there isn't a specification. Yes, we should have done a better job at creating speclets for these features, but that this point the process of creating them is the process that is going on at the dotnet/csharpstandard repo, being led by the ECMA committee. Also @BillWagner, if he wants to add anything here.

**Reply by bernd5** [reply]
Thanks for your answer.
A link to the issue(s), a sample or something similar would be great.

**Comment by BillWagner** [comment]
@bernd5 

We're tracking progress on features in two files / branches in the csharpstandard repo.
- [v6 standardization progress](https://github.com/dotnet/csharpstandard/blob/draft-v6/admin/v6-feature-tracker.md)
- [V7.x standardization progress](https://github.com/dotnet/csharpstandard/blob/draft-v6/admin/v6-feature-tracker.md)

The issues with the [type: feature](https://github.com/dotnet/csharpstandard/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+feature%22) label may also be helpful.

For the roadmap, once we've finished all the features for C# 6, I'll make a PR to change the source for the C# spec on docs.microsoft.com to the updated standard. After that, we'll start publishing from the latest versions as content is available.

As @333fred mentioned, we know we got into this state, and we're working to dig ourselves out from the technical debt incurred. It will take time, and we understand your frustration and appreciate your patience.

**Reply by bernd5** [reply]
@BillWagner thanks for your answer this helps me.

BTW.: I really like those small "speclets" by feature.

**Reply by bernd5** [reply]
I still miss for example "ref structs / span" and "ref loops" - can you help me?
Update: "ref struct" has two files "span-safety.md" and "ref-struct-and-span.md"

**Reply by BillWagner** [reply]
I may be misunderstanding what you mean by `ref loops`, but I think that's covered by [Ref local reassignment](https://docs.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-7.3/ref-local-reassignment)

**Reply by bernd5** [reply]
I mean for example: [link](https://dotnetfiddle.net/XBlv72)

**Reply by BillWagner** [reply]
Ref local reassignment is the language feature that enables that example.

**Reply by bernd5** [reply]
I mean the spec for: https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.3/ref-loops.md

See: https://docs.microsoft.com/de-de/dotnet/csharp/language-reference/keywords/foreach-in

**Reply by BillWagner** [reply]
Ahh. Yes, we do have to recreate the spec from a variety of LDM decisions.
